Server/mods/deathmatch/logic/packets/CChatEchoPacket.cpp

```diff
@@ -20,6 +20,11 @@ bool CChatEchoPacket::Write ( NetBitStreamInterface& BitStream ) const
     BitStream.Write ( m_ucBlue );
     BitStream.WriteBit ( m_bColorCoded );
 
+    // Write the client's ID
+    if ( BitStream.Version() >= 0x06B ) {
+        BitStream.Write ( GetSourceElement() ? GetSourceElement()->GetID() : NULL );

```

Comment:
It should probably be 0 rather than NULL instead, as you're not actually writing a pointer here, but the element id zero.